src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java

```diff
@@ -1433,6 +1541,342 @@ private static int validateLogicalDecodingMessageExcludeList(Configuration confi
+                                            publicationName, plugin));
+                                }
+                                else {
+                                    createOrUpdatePublicationModeFilterted(stmt, true, config);
+                                }
+                                break;
+                            default:
+                                LOGGER.trace(
+                                        "A logical publication named '{}' for plugin '{}' is already active on the server " +
+                                                "and will be used by the plugin",
+                                        publicationName, plugin);
+
+                        }
+                    }
+                }
+                conn.commit();
+                conn.setAutoCommit(true);
+            }
+            catch (SQLException e) {
+                throw new JdbcConnectionException(e);
+            }
+        }
+    }
+
+    private static void createOrUpdatePublicationModeFilterted(Statement stmt, boolean isUpdate, Configuration config) {
+        String publicationName = config.getString(PUBLICATION_NAME);
+        String tableFilterString = null;
+        String createOrUpdatePublicationStmt;
+        try {
+            Set<TableId> tablesToCapture = determineCapturedTables(config);
+            tableFilterString = tablesToCapture.stream().map(TableId::toDoubleQuotedString).collect(Collectors.joining(", "));
+            if (tableFilterString.isEmpty()) {
+                throw new DebeziumException(String.format("No table filters found for filtered publication %s", publicationName));
+            }
+            createOrUpdatePublicationStmt = isUpdate ? String.format("ALTER PUBLICATION %s SET TABLE %s;", publicationName, tableFilterString)
+                    : String.format("CREATE PUBLICATION %s FOR TABLE %s;", publicationName, tableFilterString);
+            LOGGER.info(isUpdate ? "Updating Publication with statement '{}'" : "Creating Publication with statement '{}'", createOrUpdatePublicationStmt);
+            stmt.execute(createOrUpdatePublicationStmt);
+        }
+        catch (Exception e) {
+            throw new ConnectException(String.format("Unable to %s filtered publication %s for %s", isUpdate ? "update" : "create", publicationName, tableFilterString),
+                    e);
+        }
+    }
+
+    private static Set<TableId> determineCapturedTables(Configuration config) throws Exception {
+        YugabyteDBConnection ybConnection = new YugabyteDBConnection(getJdbcConfig(config),YugabyteDBConnection.CONNECTION_GENERAL);
+        Set<TableId> allTableIds = ybConnection.readTableNames(config.getString(DATABASE_NAME), null, null, null);
+
+        Set<TableId> capturedTables = new HashSet<>();
+        RelationalTableFilters tableFilter = new RelationalTableFilters(config, new SystemTablesPredicate(), x -> x.schema() + "." + x.table());
```

Comment:
We also need to skip the tables without primary key otherwise the publication creation will fail.

Please also add a test case where we have 2 tables - one with and another without primary key. It should autocreate mode as filtered and the publication should not exist.